src/enclave/Enclave/CMakeLists.txt

```diff
@@ -15,57 +15,57 @@ set(SOURCES
 
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -nostdinc -fvisibility=hidden -fpie -fstack-protector")
 set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_CXX_FLAGS} -nostdinc++")
-set(ENCLAVE_LINK_FLAGS "-Wl,--no-undefined -nostdlib -nodefaultlibs -nostartfiles -Wl,-Bstatic -Wl,-Bsymbolic -Wl,--no-undefined -Wl,-pie,-eenclave_entry -Wl,--export-dynamic -Wl,--defsym,__ImageBase=0 -Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/Enclave.lds")
+set(ENCLAVE_LINK_FLAGS "-Wl,--no-undefined -nostdlib -nodefaultlibs -nostartfiles -Wl,-Bstatic -Wl,-Bsymbolic -Wl,--no-undefined -Wl,-pie -Wl,--export-dynamic -Wl,--defsym,__ImageBase=0")
 
 add_library(enclave_trusted SHARED ${SOURCES})
 set_property(TARGET enclave_trusted PROPERTY POSITION_INDEPENDENT_CODE OFF)
 set_target_properties(enclave_trusted PROPERTIES LINK_FLAGS ${ENCLAVE_LINK_FLAGS})
 
-find_library(TRTS_LIB sgx_trts)
-find_library(TRTS_SIM_LIB sgx_trts_sim)
-find_library(TSTDC_LIB sgx_tstdc)
-find_library(TSTDCXX_LIB sgx_tcxx)
-find_library(TKEY_EXCHANGE_LIB sgx_tkey_exchange)
-find_library(TCRYPTO_LIB sgx_tcrypto)
-find_library(SERVICE_LIB sgx_tservice)
-find_library(SERVICE_SIM_LIB sgx_tservice_sim)
-
-if(NOT "$ENV{SGX_MODE}" STREQUAL "HW")
-  message(STATUS "Building for simulated SGX")
-  set(Trts_Library_Path "${TRTS_SIM_LIB}")
-  set(Service_Library_Path "${SERVICE_SIM_LIB}")
-else()
-  message(STATUS "Building for SGX hardware")
-  set(Trts_Library_Path "${TRTS_LIB}")
-  set(Service_Library_Path "${SERVICE_LIB}")
+if ("$ENV{MODE}" STREQUAL "SIMULATE")
+  target_compile_definitions(enclave_trusted PUBLIC -DSIMULATE)
 endif()
 
-target_link_libraries(enclave_trusted -Wl,--whole-archive "${Trts_Library_Path}" -Wl,--no-whole-archive -Wl,--start-group "${TSTDC_LIB}" "${TSTDCXX_LIB}"
-  "${TKEY_EXCHANGE_LIB}" "${TCRYPTO_LIB}" "${Service_Library_Path}" -Wl,--end-group)
+target_compile_definitions(enclave_trusted PUBLIC OE_API_VERSION=2)
+
+# Need for the generated file Enclave_t.h
+target_include_directories(enclave_trusted PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
+
+target_link_libraries(enclave_trusted 
+                      openenclave::oeenclave 
+                      openenclave::oelibc 
+                      openenclave::oelibcxx
+                      openenclave::oehostsock
+		              openenclave::oehostresolver)
+
+# Sign enclave
+# add_custom_command(OUTPUT enclave/enclave.signed
+#   DEPENDS enclave enclave/helloworld.conf private.pem
+#   COMMAND openenclave::oesign sign -e $<TARGET_FILE:enclave> -c ${CMAKE_SOURCE_DIR}/enclave/helloworld.conf -k private.pem)
+
+add_custom_command(
+  COMMAND oesign sign -e $<TARGET_FILE:enclave_trusted> -c ${CMAKE_CURRENT_SOURCE_DIR}/Enclave.conf -k $ENV{OPAQUE_HOME}/private_key.pem 
+  DEPENDS enclave_trusted ${CMAKE_CURRENT_SOURCE_DIR}/Enclave.conf
+  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/enclave_trusted.signed)
 
+# TODO: Use the user-generated private key to sign the enclave code.
+# Currently we use the sample private key from the Intel SGX SDK.
```

Comment:
Is this still true?

Reply:
This has been fixed. We now use the user-generated private key to sign the enclave code.